asserts/account_key.go

```diff
@@ -43,49 +43,41 @@ func (ak *AccountKey) Since() time.Time {
 	return ak.since
 }
 
-// Until returns the time when the account key stops being valid.
+// Until returns the time when the account key stops being valid. A zero time means the key is valid forever.
 func (ak *AccountKey) Until() time.Time {
 	return ak.until
 }
 
-// PublicKeyID returns the key id (as used to match signatures to signing keys) for the account key.
-func (ak *AccountKey) PublicKeyID() string {
-	return ak.pubKey.ID()
-}
-
-// PublicKeyFingerprint returns the fingerprint of the account key.
-func (ak *AccountKey) PublicKeyFingerprint() string {
-	return ak.pubKey.Fingerprint()
+// PublicKeySHA3_384 returns the key SHAS3-384 hash used for lookup of the account key.
+func (ak *AccountKey) PublicKeySHA3_384() string {
+	return ak.pubKey.SHA3_384()
 }
 
 // isKeyValidAt returns whether the account key is valid at 'when' time.
 func (ak *AccountKey) isKeyValidAt(when time.Time) bool {
-	return (when.After(ak.since) || when.Equal(ak.since)) && when.Before(ak.until)
+	valid := when.After(ak.since) || when.Equal(ak.since)
+	if valid && !ak.until.IsZero() {
+		valid = when.Before(ak.until)
+	}
+	return valid
 }
 
 // publicKey returns the underlying public key of the account key.
 func (ak *AccountKey) publicKey() PublicKey {
 	return ak.pubKey
 }
 
-func checkPublicKey(ab *assertionBase, fingerprintName, keyIDName string) (PublicKey, error) {
+func checkPublicKey(ab *assertionBase, keyHashName string) (PublicKey, error) {
```

Comment:
Along similar lines, I'd prefer to continue referring to it as the `keyIDName`. The fundamental property is being a key ID still. Semantics vs. implementation. It'd be analogous to calling it `keyIntName` if we used ints instead of a hash. The fact it's an int is irrelevant.. the more interesting property is it being the id.